stable/database/templates/_helpers.tpl

```diff
@@ -261,7 +268,8 @@ into account any schedule overrides configured per backup group.
 Renders the name of the HotCopy CronJob
 */}}
 {{- define "hotcopy.cronjob.name" -}}
-{{ printf "%s-hotcopy-%s-%s-%s" .hotCopyType .Values.admin.domain .Values.database.name .backupGroup | trunc 52 | trimSuffix "-" }}
+{{- $name := printf "%s-hotcopy-%s-%s-%s" .hotCopyType .Values.admin.domain .Values.database.name .backupGroup -}}
```

Comment:
I'm always confused about how these values get evaluated. How can we evaluate `.hotCopyType` and also have access to the root context to be able to get `.Values.admin.domain` and `.Values.database.name`?

Reply:
Right. Lacking the ability to define custom functions (or at lest I don't know how to do it) forces us to declare named templates and use them as they were functions. The problem is that there is no way to explicitly define the input variables. Another downside is that you can't use these templates with the Helm pipeline syntax.
I found an interesting [blog post](https://itnext.io/use-named-templates-like-functions-in-helm-charts-641fbcec38da) with more tricks.